src/_pytest/_io/terminalwriter.py

```diff
@@ -218,14 +220,75 @@ def _highlight(
             return source
         else:
             try:
-                highlighted: str = highlight(
-                    source,
-                    Lexer(),
-                    TerminalFormatter(
+                # Establishes the style to be used.
+                if os.environ.get("COLORTERM", "") not in (
```

Comment:
This logic is repeated and split in several places. I suggest define a variable at the top, something like `color_mode: Literal["basic", "256", "truecolor"]`, and use that.